apollo-router/src/query_planner/tests.rs

```diff
@@ -93,9 +95,9 @@ async fn mock_subgraph_service_withf_panics_should_be_reported_as_service_closed
     };
 
     let mut mock_products_service = plugin::test::MockSubgraphService::new();
-    mock_products_service.expect_call().times(1).withf(|_| {
```

Comment:
Possibly points to a bug. The original assertion placement was correct because the original service should be consumed.

Reply:
The change is because service "creation" is now actually a clone (inside MakeSubgraphService::make). So that seems fine. The test as a whole is a bit odd and I need a bit longer to wrap my head around what it's really trying to do.